### ISSUE TITLE
Hacking with Firebug lets you buy things for free. 

### ISSUE THREAD

**Issue by justgage** [wojodesign/simplecart-js#422]
I can just change the HTML and it changes the price thus I can buy things for free. Issue? I think so.

**Comment by frozonfreak** [comment]
This has been discussed in simplecart website. Apparently this is issue with writing a fully client based shopping cart. You need to write a server check for transactions

`Is Not: Foolproof
Clever hackers can change prices of simpleCart(js) items before checkout. This is a known security flaw that exists 
with ALL javascript shopping carts. It is highly recommended that all orders be checked upon receiving or backend 
security checks be put in place (View Tutorial.)
`
Source :http://simplecartjs.org/features/

**Comment by justgage** [comment]
Yeah I sense realize that, I just found it odd that in the "try it" on the front page they redirect you directly to PayPal and the changed price in view. But yeah It's a good cart if you validate before it gets to paypal

**Comment by ninjasort** [comment]
A server check is obvious here, but what about the exposed emails of Paypal accounts. That is definitely a big security problem.

**Comment by do3cc** [comment]
The flaw already starts at paypal for allowing submitting orders without any validation information.
Since you can add arbitrary data to cart items, it is up to you to add some form of checksum or hash of your cart with some secret that stays off the site. 
Then it is up to you to validate the checksum or hash in a custom payment provider.
I do such a thing with the fancy library itsdangerous for our python web app.
Unfortunately, one can see all information sent to paypal/google checkout/amazon payments and there is no check at all, so there you can still manipulate any order. With these providers your only option is to validate payments.
There are payment providers out there that support signing of requests, like ogone. With them, the url you redirect your customers to contains a checksum with some secret. This can only be forged when one knows the secret.
